samples/dnn/yolo_detector.cpp

```diff
@@ -0,0 +1,349 @@
+/**
+ * @file yolo_detector.cpp
+ * @brief Yolo Object Detection Sample
+ * @author OpenCV team
+ */
+
+//![includes]
+#include <opencv2/dnn.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <fstream>
+#include <sstream>
+#include "iostream"
+#include "common.hpp"
+#include <opencv2/highgui.hpp>
+//![includes]
+
+using namespace cv;
+using namespace cv::dnn;
+
+void getClasses(std::string classesFile);
+void drawPrediction(int classId, float conf, int left, int top, int right, int bottom, Mat& frame);
+void yoloPostProcessing(
+    std::vector<Mat>& outs,
+    std::vector<int>& keep_classIds,
+    std::vector<float>& keep_confidences,
+    std::vector<Rect2d>& keep_boxes,
+    float conf_threshold,
+    float iou_threshold,
+    const std::string& test_name
+);
+
+std::vector<std::string> classes;
+
+
+std::string keys =
+    "{ help  h     | | Print help message. }"
+    "{ @alias      | | An alias name of model to extract preprocessing parameters from models.yml file. }"
+    "{ zoo         | models.yml | An optional path to file with preprocessing parameters }"
+    "{ device      |  0 | camera device number. }"
+    "{ input i     | | Path to input image or video file. Skip this argument to capture frames from a camera. }"
+    "{ classes     | | Optional path to a text file with names of classes to label detected objects. }"
+    "{ thr         | .5 | Confidence threshold. }"
+    "{ nms         | .4 | Non-maximum suppression threshold. }"
+    "{ mean        | 0.0 | Normalization constant. }"
+    "{ scale       | 1.0 | Normalization scalor. }"
+    "{ yolo        | None | yolo model version. }"
+    "{ padvalue    | 114.0 | padding value. }"
+    "{ paddingmode |  2 | Choose one of computation backends: "
+                         "0: resize to required input size without extra processing, "
+                         "1: Image will be cropped after resize, "
+                         "2: Resize image to the desired size while preserving the aspect ratio of original image }"
```

Comment:
I think that depends on the yolo version there is only one valid padding mode as well as other pre-processing values.

Reply:
Could you elaborate please, did not get your point

Reply:
Can you show an example of command line for YOLOv8 and YOLOv4 using this sample? I think it might be not obvious for user to collect right pre-processing flags for them.

Reply:
Default value should be 2 instead of 2.0.